pages/version/1.1.markdown

```diff
@@ -0,0 +1,350 @@
+
+Publishers should support [Conditional GET](https://fishbowl.pastiche.org/2002/10/21/http_conditional_get_for_rss_hackers/), to limit impact on their system when feed readers poll for changes.
+
+Also, to further reduce bandwidth use, publishers should provide the top-level `icon` and `favicon` URLs. When not present, feed readers will often attempt to find these things by 1) downloading the home page and scraping the HTML, and 2) making one or more requests to likely locations for these images. To reduce traffic on your server, put this info in the feed.
+
+JSON Feeds should be encoded using UTF-8 — but any encoding that’s [legal JSON](https://tools.ietf.org/html/rfc7159) is legal for JSON Feeds.
+
+Any publisher already publishing RSS and/or Atom should continue to do so. In fact, if you’re trying to decide which format (of RSS, Atom, and JSON Feed) to use, and you can do only one, pick RSS — it’s time-tested, popular, and good.
+
+## Suggestions for Feed Readers <a name="suggestions-for-feed-readers"></a>
+
+If a feed is invalid JSON, then we strongly recommend not attempting to parse it or use partial data. You can’t rely on it.
+
+There are cases, though, where a required element might not be present. In the case of an `attachment` missing a `mime_type`, you can probably deal with it fine. After all, when you download the attached file, the web browser will provide a MIME type. (And you might have been able to guess it from the file suffix.)
+
+Another case might be a malformed `date_published` that you can’t parse. You might substitute the date the reader parsed it. (Feed readers have been doing that kind of thing for many years, for bad or missing dates.)
+
+As much as we’d like to encourage good feeds, we also emphasize that this is a pragmatic format, and the final test is user experience: if an error can be recovered from without significantly harming that experience, then it’s better than just refusing to use the feed (or part of the feed) at all.
+
+That said, there is one thing we insist on: any `item` without an `id` must be discarded. We come to this from years of experience dealing with feeds in other formats where unique identifiers are optional. Without unique identifiers, it’s impossible to reliably refer to a given `item` and its changes over time, and this is terrible for user experience and becomes a source of bug reports to you. (Your users will complain of “reruns.”) In that case, you may wish to alert the user that there’s a problem with the feed, and you may also wish to contact the publisher.
+
+If an `id` is presented as a number, a JSON Feed reader should coerce it to a string. If an `id` is blank or can't be coerced to a valid string, the `item` must be discarded.
```

Comment:
I'd argue this is still too vague. There's multiple ways of coercing numbers to strings: is `3e-10` as a string `"0.0000000003"`, or is it `"3e-10"` (or `"3E-10"`) or `"0,0000000003"`? Is `10.0` `"10.0"` or `"10"`? What about `-0`? Given that feed readers need to do deduplication to identify updates, this could yield to unreliable results between implementations.

Additionally, which other types produce "valid strings"? This entirely depends on the host programming language; some might successfully coerce objects and arrays to strings. 

Given that numbers have been allowed in the past, it's too late to forbid them entirely, but I'd suggest specifying conversion rules explicitly:

1. Numbers are represented in base 10
2. Very large or very small numbers never use E notation
3. Insignificant digits are omitted, with always at least one digit to the left of the radix point
4. Negative zero is changed to positive zero
5. The radix point is represented by a U+002E FULL STOP character when required
6. JSON types other than strings and numbers must be rejected

And that's without getting into problems of precision, of course. I'm probably overly pedantic (I don't expect a system to render numbers as hexadecimal, or with no digit before the decimal point, or with a comma), but at least some of these are errors which a hurried programmer can easily make.

Reply:
These are good points, but I wonder if we add these disclaimers that people will think `id` can be a number. It really should be a string, and the 1.1 spec clarifies that, but we kept that paragraph about coercing if people want to gracefully handle feeds that have numbers (most likely a database ID that was accidentally not quoted).